### PR TITLE
[PLT-6625] Unit tests for Avatar

### PR DESCRIPTION
**Pull request by imogenem**
## Description
Unit tests for Avatar Component

## Notes
Looking for feedback on alt and src prop tests

## Where to Start
Avatar.test.jsx in components folder

### CODE REVIEW

File: packages/riipen-ui/src/components/Avatar.test.jsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    });
+
+    it("appends higher order values to default classes prop with withClass decorator", () => {
+      const wrapper = mount(<Avatar />);
+
+      expect(wrapper.find("Avatar").props().classes).toEqual([
+        "riipen",
+        "riipen-avatar"
+      ]);
+    });
```
Comment (jordanell) [inline packages/riipen-ui/src/components/Avatar.test.jsx:35]: I like this. We should do this on every component.

File: packages/riipen-ui/src/components/Avatar.test.jsx [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+      expect(
+        wrapper
+          .find(".avatar")
+          .childAt(0)
+          .hasClass("inner")
+      ).toEqual(true);
+      expect(wrapper.find("Avatar").props().alt).toEqual(altVariant);
+    });
+  });
```
Comment (jordanell) [inline packages/riipen-ui/src/components/Avatar.test.jsx:52]: I don't think this is fully correct. To me looking at the source code, the `alt` prop is only useful when `src` is also passed. If you pass both `alt` and `src`, then the `` tag has it's `alt` value set to the value of the prop` alt`.

So i _think_ what we need is a test that looks like:

```js
const wrapper = mount(<Avatar alt={altVariant} src={...} />);

// expect wrapper.find('img').props.alt to equal alt
```

Does that make sense?

File: packages/riipen-ui/src/components/Avatar.test.jsx [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+      const wrapper = mount(<Avatar alt={altVariant} />);
+
+      expect(
+        wrapper
+          .find(".avatar")
+          .childAt(0)
+          .hasClass("inner")
+      ).toEqual(true);
+      expect(wrapper.find("Avatar").props().alt).toEqual(altVariant);
```
Comment (jordanell) [inline packages/riipen-ui/src/components/Avatar.test.jsx:50]: I don't think we need this. This is that weird test again that is saying "I pass 'sum' the first arugment of "1", now tests did I pass 'sum' the first argument of 1".

File: packages/riipen-ui/src/components/Avatar.test.jsx [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    });
+  });
+
+  describe("children prop", () => {
+    it("displays given children", () => {
+      const initials = "ZZ";
+      const child = <div>{initials}</div>;
```
Comment (jordanell) [inline packages/riipen-ui/src/components/Avatar.test.jsx:57]: I don't think you need to split these like this. I think you can just do `child = <div>ZZ</div>` then test to make sure that the wrapper contains that div.

File: packages/riipen-ui/src/components/Avatar.test.jsx [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+          .find(".inner")
+          .children()
+          .contains(child)
+      ).toEqual(true);
+      expect(
+        wrapper
+          .find(".inner")
+          .childAt(0)
+          .text()
+      ).toEqual(initials);
```
Comment (jordanell) [inline packages/riipen-ui/src/components/Avatar.test.jsx:72]: If you simplify the above, you can remove this test. This test is just a subset of the one above it, so we are actually testing the same thing twice.

File: packages/riipen-ui/src/components/Avatar.test.jsx [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+  });
+
+  describe("size prop", () => {
+    it("sets valid custom size", () => {
+      const sizeVariant = "50px";
+
+      const wrapper = mount(<Avatar size={sizeVariant} />);
+
+      expect(wrapper.find("Avatar").props().size).toEqual(sizeVariant);
+    });
```
Comment (jordanell) [inline packages/riipen-ui/src/components/Avatar.test.jsx:98]: This is that weird test again. Can remove.

File: packages/riipen-ui/src/components/Avatar.test.jsx [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+
+    it("sets valid border size above 70", () => {
```
Comment (jordanell) [inline packages/riipen-ui/src/components/Avatar.test.jsx:100]: Maybe "sets correct border size when size is above 70" ? Just slightly clearer.

File: packages/riipen-ui/src/components/Avatar.test.jsx [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+
+      expect(
+        wrapper
+          .find("Avatar")
+          .instance()
+          .getBorderSize()
+      ).toEqual("4px");
+    });
+
+    it("sets valid border size below 70", () => {
```
Comment (jordanell) [inline packages/riipen-ui/src/components/Avatar.test.jsx:113]: Maybe "sets correct border size when size is below 70" ? Just slightly clearer.

File: packages/riipen-ui/src/components/Avatar.test.jsx [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+          .instance()
+          .getBorderSize()
+      ).toEqual("2px");
+    });
+  });
+
+  describe("src prop", () => {
+    it("sets image class name given custom alt and src props", () => {
+      const altVariant = "Test String";
+      const srcVariant = "/static/images/avatars/jordan-ell.png";
```
Comment (jordanell) [inline packages/riipen-ui/src/components/Avatar.test.jsx:130]: lol oh gawd this image

File: packages/riipen-ui/src/components/Avatar.test.jsx [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+    });
+  });
+
+  describe("variant prop", () => {
+    it("sets variant class name with valid custom variant", () => {
+      const variant = "rounded";
+
+      const wrapper = mount(<Avatar variant={variant} />);
+
+      expect(wrapper.find("Avatar").props().variant).toEqual(variant);
```
Comment (jordanell) [inline packages/riipen-ui/src/components/Avatar.test.jsx:150]: This is that weird test again, can remove.

File: packages/riipen-ui/src/components/Avatar.test.jsx [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+        wrapper
+          .find("Avatar")
+          .instance()
+          .getBorderSize()
+      ).toEqual("2px");
+    });
+  });
+
+  describe("src prop", () => {
+    it("sets image class name given custom alt and src props", () => {
```
Comment (jordanell) [inline packages/riipen-ui/src/components/Avatar.test.jsx:128]: Nice this is what I was trying to explain up above with the `alt` tests.

I think we don't need the `alt` here if we make the changes I proposed above. This tests should simply be "renders an img tag". See test below for how to update.

File: packages/riipen-ui/src/components/Avatar.test.jsx [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+
+      const wrapper = mount(<Avatar alt={altVariant} src={srcVariant} />);
+
+      expect(
+        wrapper
+          .find(".avatar")
+          .childAt(0)
+          .hasClass("image")
+      ).toEqual(true);
+      expect(wrapper.find("Avatar").props().src).toEqual(srcVariant);
```
Comment (jordanell) [inline packages/riipen-ui/src/components/Avatar.test.jsx:140]: This is that weird test again, can remove.

File: packages/riipen-ui/src/components/Avatar.test.jsx [hunk 13]
```diff
@@ -0,0 +1,167 @@
...
+      const srcVariant = "/static/images/avatars/jordan-ell.png";
+
+      const wrapper = mount(<Avatar alt={altVariant} src={srcVariant} />);
+
+      expect(
+        wrapper
+          .find(".avatar")
+          .childAt(0)
+          .hasClass("image")
+      ).toEqual(true);
```
Comment (jordanell) [inline packages/riipen-ui/src/components/Avatar.test.jsx:139]: This should instead be "expect wrapper.avatar to contain a "img" tag and that the img tag has its `src` prop set to match the prop we gave".

**Review by jordanell — CHANGES_REQUESTED** [review]

File: packages/riipen-ui/src/components/Avatar.test.jsx (hunk 9 above)
Comment (imogenem) [reply packages/riipen-ui/src/components/Avatar.test.jsx:130]: :}

File: packages/riipen-ui/src/components/Avatar.test.jsx (hunk 2 above)
Comment (imogenem) [reply packages/riipen-ui/src/components/Avatar.test.jsx:52]: Adjusted and renamed the test

File: packages/riipen-ui/src/components/Avatar.test.jsx (hunk 13 above)
Comment (imogenem) [reply packages/riipen-ui/src/components/Avatar.test.jsx:139]: expect(wrapper.find("Avatar").contains("img")) won't work in this case, it cannot find the img element.

As an alternative, can we use this? Is it necessary?
 expect(wrapper.find("Avatar").find("img")).toHaveLength(1);

**Review by jordanell — APPROVED** [review]
Great job!

**Pull request merged**